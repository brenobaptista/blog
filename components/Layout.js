import Head from 'next/head'
import Link from 'next/link'
import Image from 'next/image'

import styles from '../styles/Layout.module.css'
import utilStyles from '../styles/Utils.module.css'

const name = 'Breno Baptista'
export const siteTitle = 'Next.js Sample Website'

export default function Layout({ children, home }) {
  return (
    <div className={styles.container}>
      <Head>
        <link rel="icon" href="/favicon.ico" />
        <meta
          name="description"
          content="Learn how to build a personal website using Next.js"
        />
        <meta
          property="og:image"
          content={`https://og-image.now.sh/${encodeURI(
            siteTitle
          )}.png?theme=light&md=0&fontSize=75px&images=https%3A%2F%2Fassets.vercel.com%2Fimage%2Fupload%2Ffront%2Fassets%2Fdesign%2Fnextjs-black-logo.svg`}
        />
        <meta name="og:title" content={siteTitle} />
        <meta name="twitter:card" content="summary_large_image" />
      </Head>
      <header className={styles.header}>
        {home ? (
          <>
            <Image
              src="/images/profile.jpg"
              className={utilStyles.borderCircle}
              alt={name}
              width={128}
              height={128}
            />
            <h1 className={utilStyles.heading2Xl}>{name}</h1>
          </>
        ) : (
          <>
            <Link href="/">
              <a>
                <Image
                  src="/images/profile.jpg"
                  className={utilStyles.borderCircle}
                  alt={name}
                  width={96}
                  height={96}
                />
              </a>
            </Link>
            <h2 className={utilStyles.headingLg}>
              <Link href="/">
                <a className={utilStyles.colorInherit}>{name}</a>
              </Link>
            </h2>
          </>
        )}
      </header>
      <main>{children}</main>
      {!home && (
        <div className={styles.backToHome}>
          <Link href="/">
            <a>‚Üê Back to home</a>
          </Link>
        </div>
      )}
    </div>
  )
}
