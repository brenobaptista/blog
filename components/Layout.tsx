import Head from 'next/head'
import Link from 'next/link'
import Image from 'next/image'

import styles from '../styles/Layout.module.css'
import utilStyles from '../styles/Utils.module.css'

interface Props {
  children: React.ReactNode
  home?: boolean
}

const name = 'Breno Baptista'
export const siteTitle = 'Breno Baptista'

const Layout = ({ children, home }: Props) => (
  <div className={styles.container}>
    <Head>
      <link
        rel="icon"
        href="/favicon.ico"
      />
      <meta
        name="description"
        content="Breno Baptista's Portfolio"
      />
      <meta
        property="og:image"
        content={`https://og-image.now.sh/${encodeURI(
          siteTitle
        )}.png?theme=light&md=0&fontSize=75px&images=https%3A%2F%2Fassets.zeit.co%2Fimage%2Fupload%2Ffront%2Fassets%2Fdesign%2Fnextjs-black-logo.svg`}
      />
      <meta
        name="og:title"
        content={siteTitle}
      />
      <meta
        name="twitter:card"
        content="summary_large_image"
      />
    </Head>
    <header className={styles.header}>
      {home ? (
        <>
          <Image
            src="/images/profile.jpg"
            className={utilStyles.borderCircle}
            alt={name}
            width={128}
            height={128}
          />
          <h1 className={utilStyles.heading2Xl}>
            {name}
          </h1>
        </>
      ) : (
        <>
          <Link href="/">
            <a>
              <Image
                src="/images/profile.jpg"
                className={utilStyles.borderCircle}
                alt={name}
                width={96}
                height={96}
              />
            </a>
          </Link>
          <h2 className={utilStyles.headingLg}>
            <Link href="/">
              <a className={utilStyles.colorInherit}>
                {name}
              </a>
            </Link>
          </h2>
        </>
      )}
    </header>
    <main>
      {children}
    </main>
    {!home && (
      <div className={styles.backToHome}>
        <Link href="/">
          <a>
            ‚Üê Back to home
          </a>
        </Link>
      </div>
    )}
  </div>
)

export default Layout
