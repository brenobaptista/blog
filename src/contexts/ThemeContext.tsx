import { createContext } from 'react'
import { DefaultTheme } from 'styled-components'

import dracula from '@/styles/themes/dracula'

interface IThemeContext {
  theme: DefaultTheme
  toggleTheme: () => void
}

const ThemeContext = createContext<IThemeContext>({
  theme: dracula,
  toggleTheme: () => null
})

export default ThemeContext
