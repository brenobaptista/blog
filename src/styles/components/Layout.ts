import styled from 'styled-components'

export const Container = styled.div`
  max-width: 768px;
  padding: 0 16px;
  margin: 48px auto 96px;

  a {
    color: ${props => props.theme.colors.primary};
  }

  a:hover {
    text-decoration: underline;
  }

  img {
    max-width: 100%;
    display: block;
  }
`

export const Header = styled.header`
  display: flex;
  flex-direction: column;
  align-items: center;

  img {
    border-radius: 50%;
  }
`

export const Separator = styled.hr`
  border: 0;
  height: 1px;
  background: #333;
  background-image: linear-gradient(to right, #ccc, #333, #ccc);
`

export const ShortBio = styled.div`
  display: flex;

  img {
    flex: 1;
    margin-top: 28px;
    border-radius: 50%;
  }

  p {
    flex: 1;
    margin-left: 20px;
    font-style: italic;
  }
`

export const BackToHome = styled.div`
  margin: 48px 0 0;
`
