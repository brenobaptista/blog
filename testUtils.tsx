import { render } from '@testing-library/react'
import { RenderResult } from '@testing-library/react/types'
import { ThemeProvider } from 'styled-components'

import light from './src/styles/themes/light'

const Providers = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
)

const customRender = (ui: React.ReactElement, options = {}): RenderResult =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
