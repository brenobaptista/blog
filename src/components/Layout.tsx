import { ReactNode } from 'react'
import Link from 'next/link'

import Footer from '@/components/Footer'
import Sparkles from '@/components/Sparkles'
import ThemeSwitcher from '@/components/ThemeSwitcher'

import { Container, Toolbar } from '@/styles/components/Layout'
import RetroGlobalStyle from '@/styles/retro-global'

interface ILayout {
  children: ReactNode
}

const Layout = ({ children }: ILayout) => (
  <Container>
    <Toolbar>
      <Link href='/'>Breno Baptista</Link>
      <ThemeSwitcher />
    </Toolbar>
    <main>{children}</main>
    <Footer />
    <Sparkles />
    <RetroGlobalStyle />
  </Container>
)

export default Layout
