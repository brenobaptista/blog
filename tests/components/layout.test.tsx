import { render } from 'testUtils'
import Layout from '@/components/Layout'
import ThemeContext from '@/contexts/ThemeContext'

const renderWithContext = (ui, value) =>
  render(<ThemeContext.Provider value={value}>{ui}</ThemeContext.Provider>)

test('matches snapshot in dracula mode', () => {
  const toggleTheme = jest.fn()
  const theme = { mode: 'dracula' }
  const { asFragment } = renderWithContext(
    <Layout>
      <></>
    </Layout>,
    { theme, toggleTheme }
  )

  expect(asFragment()).toMatchSnapshot()
})

test('matches snapshot in retro mode for home', () => {
  const toggleTheme = jest.fn()
  const theme = { mode: 'retro' }
  const { asFragment } = renderWithContext(
    <Layout home>
      <></>
    </Layout>,
    { theme, toggleTheme }
  )

  expect(asFragment()).toMatchSnapshot()
})
