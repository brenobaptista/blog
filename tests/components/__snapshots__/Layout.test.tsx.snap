// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot in dracula mode 1`] = `
<DocumentFragment>
  .c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  font-size: 0.9rem;
}

.c11 a {
  margin: 16px;
  border-radius: 4px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c10 {
  border: 0;
  height: 1px;
  background-color: gray;
  background-image: linear-gradient(to right,gray,gray,gray);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2px;
}

.c3 {
  font-size: 0.6rem;
  font-weight: 500;
}

.c5 {
  position: absolute;
  height: 0;
  width: 0;
  visibility: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  cursor: url('/images/retro/cursor/pointer.cur'),pointer;
  border-radius: 56px;
  width: 56px;
  height: 28px;
  background-color: orange;
}

.c9 {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  border-radius: 24px;
  width: 24px;
  height: 24px;
  background-color: black;
  -webkit-transition: 0.2s all ease-in-out;
  transition: 0.2s all ease-in-out;
}

.c4:checked + .c6 .c8 {
  left: calc(100% - 2px);
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c0 {
  max-width: 768px;
  padding: 16px;
  margin: 16px auto;
  background-color: black;
  border: 1px solid gray;
  border-radius: 8px;
  font-family: monospace;
}

.c0 img {
  max-width: 100%;
  display: block;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 32px;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1;
  padding: 8px;
  border-radius: 0 0 8px 8px;
  background-color: black;
  -webkit-backdrop-filter: blur(4px);
  backdrop-filter: blur(4px);
}

.c1 a {
  font-size: 1.4rem;
  font-weight: bold;
}

@media screen and (max-width:360px) {
  .c11 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    text-align: center;
  }
}

@media screen and (max-width:768px) {
  .c0 {
    margin: 8px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        href="/"
      >
        Breno Baptista
      </a>
      <div
        class="c2"
      >
        <label
          class="c3"
          for="checkbox"
        >
          Retro mode
        </label>
        <input
          class="c4 c5"
          id="checkbox"
          type="checkbox"
        />
        <label
          aria-label="Change the visual styles"
          class="c6 c7"
          data-testid="label-checkbox"
          for="checkbox"
          tabindex="0"
        >
          <span
            class="c8 c9"
          />
        </label>
      </div>
    </div>
    <main />
    <footer>
      <hr
        class="c10"
      />
      <nav
        class="c11"
      >
        <div
          class="c12"
        >
          <a
            href="mailto:me@brenobaptista.com"
          >
            Email
          </a>
        </div>
        <div
          class="c12"
        >
          <a
            href="https://github.com/brenobaptista/"
            rel="noreferrer noopener"
            target="_blank"
          >
            GitHub
          </a>
        </div>
        <div
          class="c12"
        >
          <a
            href="/rss.xml"
          >
            RSS
          </a>
        </div>
        <div
          class="c12"
        >
          <a
            href="/resume"
          >
            Résumé
          </a>
        </div>
      </nav>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot in retro mode for home 1`] = `
<DocumentFragment>
  .c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  font-size: 0.9rem;
}

.c11 a {
  margin: 16px;
  border-radius: 4px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c10 {
  border: 0;
  height: 1px;
  background-color: gray;
  background-image: linear-gradient(to right,gray,gray,gray);
}

.c13 {
  margin: 24px 0;
  font-size: 0.9rem;
  text-align: center;
}

.c13 a {
  margin: 0 8px;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 8px 0 16px;
}

.c15 {
  text-align: center;
  font-size: 0.8rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2px;
}

.c3 {
  font-size: 0.6rem;
  font-weight: 500;
}

.c5 {
  position: absolute;
  height: 0;
  width: 0;
  visibility: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  cursor: url('/images/retro/cursor/pointer.cur'),pointer;
  border-radius: 56px;
  width: 56px;
  height: 28px;
  background-color: orange;
}

.c9 {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  border-radius: 24px;
  width: 24px;
  height: 24px;
  background-color: black;
  -webkit-transition: 0.2s all ease-in-out;
  transition: 0.2s all ease-in-out;
}

.c4:checked + .c6 .c8 {
  left: calc(100% - 2px);
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c0 {
  max-width: 768px;
  padding: 16px;
  margin: 16px auto;
  background-color: black;
  border: 1px solid gray;
  border-radius: 8px;
  font-family: monospace;
}

.c0 img {
  max-width: 100%;
  display: block;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 32px;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1;
  padding: 8px;
  border-radius: 0 0 8px 8px;
  background-color: black;
  -webkit-backdrop-filter: blur(4px);
  backdrop-filter: blur(4px);
}

.c1 a {
  font-size: 1.4rem;
  font-weight: bold;
}

@media screen and (max-width:360px) {
  .c11 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    text-align: center;
  }
}

@media screen and (max-width:768px) {
  .c0 {
    margin: 8px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        href="/"
      >
        Breno Baptista
      </a>
      <div
        class="c2"
      >
        <label
          class="c3"
          for="checkbox"
        >
          Retro mode
        </label>
        <input
          checked=""
          class="c4 c5"
          id="checkbox"
          type="checkbox"
        />
        <label
          aria-label="Change the visual styles"
          class="c6 c7"
          data-testid="label-checkbox"
          for="checkbox"
          tabindex="0"
        >
          <span
            class="c8 c9"
          />
        </label>
      </div>
    </div>
    <main />
    <footer>
      <hr
        class="c10"
      />
      <nav
        class="c11"
      >
        <div
          class="c12"
        >
          <a
            href="mailto:me@brenobaptista.com"
          >
            Email
          </a>
        </div>
        <div
          class="c12"
        >
          <a
            href="https://github.com/brenobaptista/"
            rel="noreferrer noopener"
            target="_blank"
          >
            GitHub
          </a>
        </div>
        <div
          class="c12"
        >
          <a
            href="/rss.xml"
          >
            RSS
          </a>
        </div>
        <div
          class="c12"
        >
          <a
            href="/resume"
          >
            Résumé
          </a>
        </div>
      </nav>
      <hr
        class="c10"
      />
      <div
        class="c13"
      >
        <a
          href="https://hotlinewebring.club/brenobaptista/previous"
          rel="noreferrer noopener"
          target="_blank"
        >
          ←
        </a>
        <a
          href="https://hotlinewebring.club/"
          rel="noreferrer noopener"
          target="_blank"
        >
          Hotline Webring
        </a>
        <a
          href="https://hotlinewebring.club/brenobaptista/next"
          rel="noreferrer noopener"
          target="_blank"
        >
          →
        </a>
      </div>
      <div
        class="c14"
      >
        <img
          alt="Bookmark this page"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Sign my guestbook"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Best viewed with anything"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Free speech forever"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Nuke IE"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Anarchy now"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Best viewed with eyes"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Tux - Linux penguin"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Geocities"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Internet privacy"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Wiby search engine"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="I survived Y2K"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Email me"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="Best viewed with a monitor"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="I wish it was 2001"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="No code and run"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
      <div
        class="c15"
      >
        © 2020-2025 Breno Baptista
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
